src/main/java/com/palantir/common/streams/MoreStreams.java

```diff
@@ -57,6 +59,30 @@
     }
 
     /**
+     * Given a {@code Stream<ListenableFuture<U>>}, this function will return a blocking stream of the completed
+     * futures in the same order as the source, looking at most {@code maxParallelism} futures ahead in the stream.
+     */
+    public static <T extends ListenableFuture<U>, U> Stream<T> withParallelLookahead(
```

Comment:
After honing the JavaDoc, I think `blockingStreamWithParallelism` works better

Reply:
done